### PR TITLE
cleanup(rules): initial transitions of some sandbox rules to incubating

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area rules

> /area registry

> /area build

> /area documentation

**What this PR does / why we need it**:

Related to Phase 2 of rules maturity framework implementation.

"Re-audit the first round of rules tagged with maturity_sandbox and consider transitioning some to maturity_incubating, with a maximum of 5 rules per PR to enable focused discussions."

@jasondellaluce and @LucaGuerra you will see my suggestion in the yaml changes. Re-audited each rule and at the moment I would only recommend to elevate those 3 rules. Would you have additional recommendations?

There will a final re-audit at the end.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: rules/falco_rules.yaml [hunk 1]
```diff
@@ -1607,7 +1607,7 @@
     Namespace change (setns) by unexpected program (user=%user.name user_loginuid=%user.loginuid command=%proc.cmdline pid=%proc.pid
     parent=%proc.pname %container.info container_id=%container.id image=%container.image.repository:%container.image.tag)
   priority: NOTICE
-  tags: [maturity_sandbox, host, container, process, mitre_privilege_escalation, mitre_lateral_movement, T1611]
+  tags: [maturity_incubating, host, container, process, mitre_privilege_escalation, mitre_lateral_movement, T1611]
```
Comment (incertum) [inline rules/falco_rules.yaml:1610]: "Change thread namespace" relevant for the broader attack category of container escapes.

File: rules/falco_rules.yaml [hunk 2]
```diff
@@ -2741,7 +2741,7 @@
     command=%proc.cmdline pid=%proc.pid container_id=%container.id container_name=%container.name image=%container.image.repository:%container.image.tag)
   priority:
     NOTICE
-  tags: [maturity_sandbox, host, container, process, users, mitre_persistence, T1548.001]
+  tags: [maturity_incubating, host, container, process, users, mitre_persistence, T1548.001]
```
Comment (incertum) [inline rules/falco_rules.yaml:2744]: "Set Setuid or Setgid bit" relevant for broader attack category of privilege escalations.

File: rules/falco_rules.yaml [hunk 3]
```diff
@@ -2997,7 +2997,7 @@
      image=%container.image.repository namespace=%k8s.ns.name
      fd.rip.name=%fd.rip.name fd.lip.name=%fd.lip.name fd.cip.name=%fd.cip.name fd.sip.name=%fd.sip.name)
   priority: WARNING
-  tags: [maturity_sandbox, container, network, mitre_discovery, T1046]
+  tags: [maturity_incubating, container, network, mitre_discovery, T1046]
```
Comment (incertum) [inline rules/falco_rules.yaml:3000]: "Network Connection outside Local Subnet" was tagged as one of the initially relevant PCI DSS rule in https://falco.org/blog/falco-pci-controls/.

**Review by LucaGuerra — APPROVED** [review]
That is great, thank you for doing this! :pray:

**Pull request merged**